### PR TITLE
ENH: Providing ITKv5 updates for C++11

### PR DESCRIPTION
**Pull request by hjmjohnson**
This pull request incorporates code changes
that happened in parallel with updating the
core ITK code to ITKv5 for C++11 conformity.

### CODE REVIEW

File: include/itkRieszFrequencyFunction.hxx [hunk 1]
```diff
@@ -129,9 +129,8 @@ RieszFrequencyFunction< TFunctionValue, VImageDimension, TInput >
   const SetType &allIndices = this->m_Indices;
 
   OutputComponentsType out;
-  for(typename SetType::const_iterator it = allIndices.begin(); it != allIndices.end(); ++it)
+  for(auto indice : allIndices)
```
Comment (dzenanz) [inline include/itkRieszFrequencyFunction.hxx]: indice -> index

File: include/itkRieszRotationMatrix.hxx [hunk 2]
```diff
@@ -97,29 +97,29 @@ typename RieszRotationMatrix< T, VImageDimension >::IndicesMatrix
...
-  typedef std::vector< unsigned int >     IndicesArrayType;
-  typedef std::vector< IndicesArrayType > IndicesVector;
-  typedef std::vector< IndicesVector >    IndicesMatrixRow;
-  typedef std::vector< IndicesMatrixRow > IndicesMatrix;
-  IndicesMatrix allIndicesPairs( this->m_Components, // number of rows
-    IndicesMatrixRow(this->m_Components, // number of columns
-      IndicesVector(2, // pair of indices
-        IndicesArrayType(VImageDimension) // dimension of the indices.
-        )));
+  typedef std::vector< unsigned int >               LocalIndicesArrayType;
```
Comment (dzenanz) [inline include/itkRieszRotationMatrix.hxx]: Wait, what happened to `typedef` -> using change?

File: include/itkRieszRotationMatrix.hxx [hunk 3]
```diff
@@ -97,29 +97,29 @@ typename RieszRotationMatrix< T, VImageDimension >::IndicesMatrix
...
-  // Populate IndicesMatrix.
+  // Populate LocalIndicesMatrix.
     {
     unsigned int ind_i = 0;
-    for ( typename SetType::const_iterator itN = allIndices.begin(); itN != allIndices.end(); ++itN )
+    for ( auto itN = allIndices.begin(); itN != allIndices.end(); ++itN )
       {
       unsigned int ind_j = 0;
-      for ( typename SetType::const_iterator itM = allIndices.begin(); itM != allIndices.end(); ++itM )
+      for (const auto & allIndice : allIndices)
```
Comment (dzenanz) [inline include/itkRieszRotationMatrix.hxx]: allIndice -> index

File: include/itkRieszRotationMatrix.hxx [hunk 4]
```diff
@@ -192,41 +192,37 @@ RieszRotationMatrix< T, VImageDimension >
...
-        kValidIndices.push_back( IndicesVector(1, *itValid0) );
+        kValidIndices.push_back( IndicesVector(1, itValid0) );
         }
 
       unsigned int combineIterations = VImageDimension - 1;
       for ( unsigned int combineIndex = 0; combineIndex < combineIterations; ++combineIndex )
         {
         ValidIndicesType tmpValidIndices; // store new valid indices.
-        for ( unsigned int validKIndex = 0; validKIndex < kValidIndices.size(); ++validKIndex )
+        for (auto & kValidIndice : kValidIndices)
```
Comment (dzenanz) [inline include/itkRieszRotationMatrix.hxx]: kValidIndice -> kValidIndex

File: include/itkRieszRotationMatrix.hxx [hunk 5]
```diff
@@ -192,41 +192,37 @@ RieszRotationMatrix< T, VImageDimension >
...
-          for ( typename SetType::const_iterator itKIni = kValidInitialIndices[combineIndex + 1].begin();
-                itKIni != kValidInitialIndices[combineIndex + 1].end();
-                ++itKIni )
+          for (const auto & itKIni : kValidInitialIndices[combineIndex + 1])
             {
             // ---verbose sum---
-            IndicesArrayType sumKIndices = *itKIni;
-            for ( unsigned int kIndice = 0; kIndice < kValidIndices[validKIndex].size(); ++kIndice )
+            IndicesArrayType sumKIndices = itKIni;
+            for (auto & kIndice : kValidIndice)
```
Comment (dzenanz) [inline include/itkRieszRotationMatrix.hxx]: all *indice -> *index

File: include/itkRieszFrequencyFunction.hxx (hunk 1 above)
Comment (hjmjohnson) [reply include/itkRieszFrequencyFunction.hxx]: Done.

File: include/itkRieszRotationMatrix.hxx (hunk 2 above)
Comment (hjmjohnson) [reply include/itkRieszRotationMatrix.hxx]: @dzenanz 
Those changes were not applied to the remote modules.  That is on the low priority todo list.

Comment (dzenanz) [reply include/itkRieszRotationMatrix.hxx]: OK, thanks.

**Comment by hjmjohnson** [comment]
@dzenanz OK this is going to be a lot of tedious work.  The remote modules are really a pain to deal with.  We need a quick "BUILD THE CURRENT HEAD" mechanism so that the hash values are not constantly refusing to build while updating and testing.

File: include/itkExpandWithZerosImageFilter.h [hunk 6]
```diff
@@ -77,23 +77,23 @@ class ExpandWithZerosImageFilter:
...
   /** Inherit some types from superclass. */
-  typedef typename Superclass::InputImageType  InputImageType;
-  typedef typename Superclass::OutputImageType OutputImageType;
-  typedef typename OutputImageType::PixelType  OutputPixelType;
-  typedef typename InputImageType::Pointer     InputImagePointer;
-  typedef typename OutputImageType::Pointer    OutputImagePointer;
+  using InputImageType = typename Superclass::InputImageType ;
+  using OutputImageType = typename Superclass::OutputImageType;
+  using OutputPixelType = typename OutputImageType::PixelType ;
+  using InputImagePointer = typename InputImageType::Pointer    ;
```
Comment (dzenanz) [inline include/itkExpandWithZerosImageFilter.h]: In quite a few instances, semicolon is separated by a significant white space from the declaration.

**Comment by hjmjohnson** [comment]
@dzenanz Fixed.  Thanks for catching this oversight.  I missed the final regex fix that cleans that up.

**Pull request merged**